### PR TITLE
Adds code coverage workflow and badge.

### PR DESCRIPTION
**Pull request by claravanstaden**
@ralexstokes to finish this PR the ssz-rs repo would need the [CodeCov app to be added](https://docs.codecov.com/docs#prerequisites).

### CODE REVIEW

**Comment by ralexstokes** [comment]
I've added the code cov app for this repo. And generally thumbs up on this approach

it looks like this PR uses tarpaulin, which I haven't used before but don't see any issues yet -- ill go do some more research but if you want to send over any resources that would be appreciated!

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -60,3 +60,18 @@ jobs:
...
         run: cargo +nightly clippy --all-targets --all-features --verbose -- -D warnings
+
+  coverage:
+    runs-on: ubuntu-latest
+    container:
+      image: xd009642/tarpaulin:develop-nightly
+      options: --security-opt seccomp=unconfined
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (ralexstokes) [inline .github/workflows/rust.yml:71]: just for uniformity
```suggestion
    steps:
      - name: Checkout sources
        uses: actions/checkout@v3
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -60,3 +60,18 @@ jobs:
...
+
+      - name: Generate code coverage
+        run: cargo +nightly tarpaulin --verbose --all-features --workspace --timeout 120 --out Xml
+
+      - name: Upload to codecov.io
+        uses: codecov/codecov-action@v2
```
Comment (ralexstokes) [inline .github/workflows/rust.yml:77]: looks like a newer one is available

```suggestion
        uses: codecov/codecov-action@v3
```

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -60,3 +60,18 @@ jobs:
...
+    runs-on: ubuntu-latest
+    container:
+      image: xd009642/tarpaulin:develop-nightly
+      options: --security-opt seccomp=unconfined
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Generate code coverage
+        run: cargo +nightly tarpaulin --verbose --all-features --workspace --timeout 120 --out Xml
```
Comment (ralexstokes) [inline .github/workflows/rust.yml:74]: I'd like to try to keep this option

```suggestion
        run: cargo +nightly tarpaulin --engine llvm --verbose --all-features --workspace --timeout 120 --out Xml
```

**Review by ralexstokes — APPROVED** [review]
did some research and this looks like a good way forward for now!

I left some comments about updating some things but otherwise looks good to me

**Comment by claravanstaden** [comment]
@ralexstokes thanks for reviewing! I applied the changes. 😄

**Pull request merged**